ci/docker/runtime_functions.sh

```diff
@@ -715,7 +715,7 @@ build_ubuntu_gpu_mkldnn() {
     set -ex
     cd /work/build
     cmake \
-        -DCMAKE_BUILD_TYPE="RelWithDebInfo" \
```

Comment:
Why does this PR disable `-WError` checks?

Reply:
Because otherwise the size of `libmxnet.a` will go beyond `4 GB` and `ar` will fail ([error log](https://discuss.mxnet.io/t/mxnet-ci-failure-file-truncated/5749), [open issue on file truncation error](https://github.com/apache/incubator-mxnet/issues/14263)).

Reply:
I also noticed that `Release` mode has already been used in other build settings such as `build_ubuntu_gpu_cmake `, `build_ubuntu_gpu_cmake_no_rtc `, `build_ubuntu_gpu_cmake_no_tvm_op` etc.

Reply:
Yes, we use release mode in some builds and dev mode in others to ensure both builds keep working.

Why does this PR increase the binary size so significantly?

Reply:
It does not increase significantly. The size of `libmxnet.a` without this change is already around `4 GB`.

Reply:
Ok. This may require some change to CI as the old toolchain used on CI may not support the situation.

I didn't face issues with `> 4GB` `libmxnet.a` locally:

```
-rw-rw-r--  1 ubuntu ubuntu 6.6G Mar 14 03:15 libmxnet.a
-rwxrwxr-x  1 ubuntu ubuntu 3.1G Mar 14 03:16 libmxnet.so
```